### PR TITLE
OHRM5X-111: Stop deleting the already in use termination Record

### PR DESCRIPTION
**Pull request by ireshe**
changed the termination Record screen,
changes the DAO layer and added reasonInUse function,
add validation rule to the API

### CODE REVIEW

**Review by Super-Chama — APPROVED** [review]

File: symfony/plugins/orangehrmPimPlugin/Api/TerminationReasonConfigurationAPI.php [hunk 1]
```diff
@@ -141,11 +141,23 @@ public function delete(): EndpointResult
 
     /**
      * @inheritDoc
+     * @throws DaoException
      */
     public function getValidationRuleForDelete(): ParamRuleCollection
     {
+        $isinuse = $this->getTerminationReasonConfigurationService()->reasonInUse();
```
Comment (RajithaKumara) [inline symfony/plugins/orangehrmPimPlugin/Api/TerminationReasonConfigurationAPI.php:148]: refactor variable and method
```php
$reasonIdsInUse = $this->getTerminationReasonConfigurationService()->getReasonIdsInUse();
```

File: symfony/plugins/orangehrmPimPlugin/Controller/TerminationReasonController.php [hunk 2]
```diff
@@ -21,12 +21,26 @@
...
 use OrangeHRM\Core\Controller\AbstractVueController;
 use OrangeHRM\Core\Vue\Component;
+use OrangeHRM\Core\Vue\Prop;
+use OrangeHRM\Pim\Service\TerminationReasonConfigurationService;
 
 class TerminationReasonController extends AbstractVueController
 {
+    protected ?TerminationReasonConfigurationService $terminationReasonService = null;
+
+    protected function getTerminationReasonService(): TerminationReasonConfigurationService
```
Comment (RajithaKumara) [inline symfony/plugins/orangehrmPimPlugin/Controller/TerminationReasonController.php:31]: doc comment

File: symfony/plugins/orangehrmPimPlugin/Controller/TerminationReasonController.php [hunk 3]
```diff
@@ -21,12 +21,26 @@
...
+        if (!$this->terminationReasonService instanceof TerminationReasonConfigurationService) {
+            $this->terminationReasonService = new TerminationReasonConfigurationService();
+        }
+        return $this->terminationReasonService;
+    }
+
     public function init(): void
     {
         $component = new Component('termination-reason-list');
+        $reasonesInUse = $this->getTerminationReasonService()->reasonInUse();
```
Comment (RajithaKumara) [inline symfony/plugins/orangehrmPimPlugin/Controller/TerminationReasonController.php:42]: refactor `reasonInUse`

**Review by RajithaKumara — CHANGES_REQUESTED** [review]

**Comment by RajithaKumara** [comment]
Closing since moved to #831

**Pull request closed without merging**